src/components/MediaSettings/VideoBackgroundEditor.vue

```diff
@@ -36,43 +39,76 @@
 			<Blur :size="20" />
 			{{ t('spreed', 'Blur') }}
 		</button>
-		<!-- hide custom background for now -->
-		<button v-if="false" key="upload" class="background-editor__element">
-			<ImagePlus :size="20" />
-			{{ t('spreed', 'Custom') }}
-		</button>
-		<button v-for="path in backgrounds"
-			:key="path"
-			aria-label="TODO: add image names as aria labels"
-			class="background-editor__element"
-			:class="{'background-editor__element--selected': selectedBackground === path}"
-			:style="{
-				'background-image': 'url(' + path + ')'
-			}"
-			@click="handleSelectBackground(path)">
-			<CheckBold v-if="selectedBackground === path"
-				:size="40"
-				fill-color="#fff" />
-		</button>
+		<template v-if="hasBackgroundsCapability">
+			<template v-if="canUploadBackgrounds">
+				<button class="background-editor__element"
+					@click="clickImportInput">
+					<Upload :size="20" />
+					{{ t('spreed', 'Upload') }}
+				</button>
+				<button class="background-editor__element"
+					:class="{'background-editor__element--selected': isCustomBackground }"
+					@click="openPicker">
+					<Folder :size="20" />
+					{{ t('spreed', 'Files') }}
+				</button>
+			</template>
+			<button v-for="path in predefinedBackgroundsURLs"
+				:key="path"
+				aria-label="TODO: add image names as aria labels"
+				class="background-editor__element"
+				:class="{'background-editor__element--selected': selectedBackground === path}"
+				:style="{
+					'background-image': 'url(' + path + ')'
+				}"
+				@click="handleSelectBackground(path)">
+				<CheckBold v-if="selectedBackground === path"
+					:size="40"
+					fill-color="#fff" />
+			</button>
+		</template>
+		<!--native file picker, hidden -->
+		<input v-show="false"
```

Comment:
```suggestion
		<input class="hidden-visually"
```
`v-show="false"` is basically a `display: none`, which may cause some troubles. suggestion is the same as used in NewMessageForm